tests/integration/test_k8s.py

```diff
@@ -44,7 +45,19 @@ async def ready_nodes(k8s, expected_count):
 
 
 @pytest.mark.abort_on_fail
-async def test_nodes_ready(kubernetes_cluster):
+async def test_nodes_ready(kubernetes_cluster: model.Model):
     """Deploy the charm and wait for active/idle status."""
     k8s = kubernetes_cluster.applications["k8s"]
     await ready_nodes(k8s.units[0], 3)
+
+
+@pytest.mark.abort_on_fail
+async def test_remove_units(kubernetes_cluster: model.Model):
+    """Deploy the charm and wait for active/idle status."""
+    k8s: application.Application = kubernetes_cluster.applications["k8s"]
+    await ready_nodes(k8s.units[0], 3)
+
+    kubernetes_cluster.destroy_unit(k8s.units[1])
+    await ready_nodes(k8s.units[0], 2)
+    k8s.destroy_relation("k8s", "k8s-worker:cluster")
+    await ready_nodes(k8s.units[0], 1)
```

Comment:
This will not work. 
The relation will be departed but ofc it takes some time for the event to be processed. 
Is there a clean way in Juju to wait for this?